modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/ObjectActionContainer.tsx

```diff
@@ -173,11 +180,17 @@ export function ObjectActionContainer({
 			title={Liferay.Language.get('new-action')}
 		>
 			<ClayTabs>
-				{TABS.map((label, index) => (
+				{TABS.map(({key, label}, index) => (
 					<ClayTabs.Item
 						active={activeIndex === index}
 						key={index}
-						onClick={() => setActiveIndex(index)}
+						onClick={() => {
+							setActiveIndex(index);
+
+							if (key === 'action-builder') {
+								setReloadFDS(true);
```

Comment:
Hey @rebsilva, Instead of creating the state here, I would prefer to create the state only in the component that will use it, and here just fire and event, the catch there avoiding unnecessary prop drilling like this:

```ts
Liferay.fire('reloadFDS');
```

And I would prefer to wrap just the PredefinedValueTable in the ActionContainer component for loading rather than the entire ActionContainer what do you think?

```tsx
const [reloadFDS, setReloadFDS] = useState(false);

...

	const handleReloadFDS = () => {
		setReloadFDS(true);

		setTimeout(() => setReloadFDS(false), 200);
	}

	useEffect(() => {
		Liferay.on('reloadFDS', handleReloadFDS);

		return () => {
			Liferay.detach('reloadFDS', handleReloadFDS);
		};
	}, []);
```

Reply:
i think it make sense! thanks for your suggestion!